lapce-app/src/command.rs

```diff
@@ -524,6 +524,16 @@ pub enum InternalCommand {
     OpenFileChanges {
         path: PathBuf,
     },
+    StartRenamePath {
+        path: PathBuf,
+    },
+    TestRenamePath {
+        new_path: PathBuf,
+    },
+    FinishRenamePath {
+        current_path: PathBuf,
+        new_path: PathBuf,
+    },
```

Comment:
Why do we need those? It should be just `Rename` command

Reply:
`StartRenamePath` opens the text box in the file explorer to allow the user to enter a new path, but doesn't perform a rename.

`TestRenamePath` queries `lapce-proxy` to ask if the rename will succeed, but doesn't perform a rename. It's used to provide the user feedback on each keypress if, for example, they have entered a path that already exists.

`FinishRenamePath` attempts to perform the rename.

Reply:
We don't need to ask proxy for validation, we already have tree in lapce-app

Reply:
The user may have entered a path like `foo/.././bar`, or a path containing symbolic links. `lapce-proxy` can check whether a path is valid for the rename by querying the actual filesystem, which is more robust than trying to handle edge cases without touching the filesystem.

Reply:
I've thought about this a bit more, and think that paths containing `.`, `..`, `//` and symbolic links could all be handled using the tree in `lapce-app`, and in any case if some edge case isn't handled then that may be okay, as `TestRenamePath` is only supposed to provide a best effort guess of whether a rename will succeed, not a guarantee.

I don't think that renames to paths outside the workspace (for example, renaming `foo` located at workspace root to `../foo`) could be validated without querying `lapce-proxy` though, as the tree only knows about paths in the workspace.

Reply:
Moving files outside workspace should be forbidden operation.